reproducible/prep-nix-build.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+#
+# Setup a host for docker builds using nix. This assumes nix has been installed
+# Setting up the nix build on the host as opposed to completely restricting the build
+# infrastructure means we don't need to set up libsgx infrastructure in the container
+# and can instead run tests separately. This is desirable as many build machines don't support
+# SGX, or at least don't have it enabled. 
+#
+# prep-nix-build.sh $nixpkgs_dest
+#
+#
+# First clone the nixpkgs repo. With a little discipline we can 
+# keep all package accesses on this machine.
+#
+export NIX_PKGS_REPO=https://github.com/yakman2020/nixpkgs.git
```

Comment:
https://github.com/yakman2020/nixpkgs.git acc-test2 branch is thousands of commits ahead of NixOS:master. What's the critical difference between https://github.com/yakman2020/nixpkgs.git acc-test2 and NixOS:master? How does the difference affect the reproducible build process?

Reply:
So long as the branch is the same, the output is the same.

Reply:
@yakman2020 , what's the difference between https://github.com/yakman2020/nixpkgs.git acc-test2 and NixOS:master? If NixOS:master is used instead, what happens?

Reply:
NixOS master doesn't work.  Everything after release-20.03 (which acc-test2 is based on) is based on glibc-2.32 which is only compatible with ubuntu 20.10 and fedora 33. 

Everything before 20.03 is EOL.

Reply:
The acc-test2 branch is based on release-20.03. There is an openenclave-sdk directory but I have not tried doing a full hydra build until the basic pr is done. The reason for the many commits is that I had to merge with 20.03 when 20.09 turned out to be unusable.  I could rebuild the branch from release-20.03 and eliminate the commits.